src/distilabel/steps/tasks/sentence_transformers.py

```diff
@@ -337,3 +364,57 @@ def format_output(
             }
 
         return {"positive": groups[0].strip()}
+
+    @override
+    def get_structured_output(self) -> Dict[str, Any]:
+        """Creates the json schema to be passed to the LLM, to enforce generating
+        a dictionary with the output which can be directly parsed as a python dictionary.
+
+        Returns:
+            JSON Schema of the response to enforce.
+        """
+        from distilabel.llms import InferenceEndpointsLLM
+        from distilabel.llms.base import AsyncLLM
+
+        if self.triplet:
+            schema = {
+                "properties": {
+                    "positive": {"title": "Positive", "type": "string"},
+                    "negative": {"title": "Negative", "type": "string"},
+                },
+                "required": ["positive", "negative"],
+                "title": "Schema",
+                "type": "object",
+            }
+        else:
+            schema = {
+                "properties": {"positive": {"title": "Positive", "type": "string"}},
+                "required": ["positive"],
+                "title": "Schema",
+                "type": "object",
+            }
+
+        # To determine instructor or outlines format
+        if isinstance(self.llm, AsyncLLM) and not isinstance(
+            self.llm, InferenceEndpointsLLM
+        ):
+            return {"schema": schema}
+
+        return {"format": "json", "schema": schema}
```

Comment:
We should move this logic so it doesn't need to be coded in every task. Maybe we can just return the `schema` in the `get_structured_output` method, and then move this condition to the `Task.load` method.